readme.md

```diff
@@ -346,6 +346,36 @@ import ky from 'ky';
 })();
 ```
 
+##### parseJson
+
+Type: `Function`\
+Default: `JSON.parse()`
+
+User-defined JSON-parsing function.
+
+Use-cases:
+1. Parse JSON via the [`bourne` package](https://github.com/hapijs/bourne) to protect from prototype pollution.
+2. Parse JSON with [`reviver` option of `JSON.parse()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse).
+
+```js
+import ky from 'ky';
+import bourne from '@hapijs/bourne';
+
+(async () => {
+	const parsed = await ky('https://example.com', {
+		parseJson: text => bourne(text)
+	}).json();
+})();
+```
+
+Please note that this option only works when using it like `ky(…).json()` (like in the example above).
+It won't work if you first get the response object and then call `.json()` on it:
```

Comment:
Why can't we make this work?

Reply:
@sindresorhus unlike the promise that Ky returns and its `promise.json()` method, we don't construct the response object, as it's the return value from `fetch()`. As far as I know, we can't easily change the behavior of its `response.json()`. I suppose we may be able to monkey patch it. How would you like to approach that?

Reply:
I think I'm with @sholladay on that. I think that monkey-patching the native `fetch.json` is a bit too far to go.

I don't have the stats of how often people use ky's json vs ky.response.json, but IMO the big selling point of ky is (among other things) these shorthand methods. There should be strong reasons to not use them AND also use parseJson option in the same time.

Reply:
We could just create a `Proxy` wrapper.

Reply:
Doesn't seem to work, even with an empty / transparent proxy (tried in Firefox 76):

```js
const response = new Response(JSON.stringify({foo:'bar'}));
const proxy = new Proxy(response, {});
proxy.json().then(console.log);
// TypeError: 'json' called on an object that does not implement interface Response.
```

Reply:
I think you actually have to implement the `apply` hook and use `Reflect.apply` correctly.

Reply:
Looking at this with a fresh pair of eyes, I think it may be sufficient to overwrite the `json` method of the `response` object on this line: https://github.com/sindresorhus/ky/blob/2c5d6dda8ac889048408a337c3f5d068769981b2/index.js#L330

We would do something like this...

```js
if (this._options.parseJson) {
    response.json = this._options.parseJson
}
return response;
```

Earlier, I was thinking that it might not be enough due to usage of `response.clone()`, as the custom `json` function would not be cloned. However, after looking at the places we use response cloning, I don't think it's actually a problem.

Reply:
Right. Let's try that. 👍

^ @kirillgroshkov